trevorskaggs/shelterly
# Adds customizable and extensible icons for default species type images

## overview
In this feature we have a small, but customizable set of "Shelterly Icons", built in as a component library. On the surface we are simply replacing the default "image not found" image for animals with specific species type defaults.

before this feature, a list of animals with no photos would look like so:
<img width="1043" alt="Screen Shot 2022-05-22 at 9 06 23 PM" src="https://user-images.githubusercontent.com/1508574/169725335-eb925e19-534f-4167-8794-7e64e7d4b788.png">

the same list fo animals with this new feature:
<img width="1044" alt="Screen Shot 2022-05-22 at 9 06 53 PM" src="https://user-images.githubusercontent.com/1508574/169725379-cf5101db-32af-4c42-9697-93982ca31644.png">

## feature features
The AnimalCoverImage component is being used to display an animal's photo, or the species default icon if no photo is defined. The SpeciesIcon component is responsible for rendering the proper icon. The ShIcon component handles rendering the svg, which is how the underlying icons are imported into react. The ShIcon component along with svg's is where the extensibility lies, for when we want o use custom icons in other areas of the app.

## quick testing steps
- login
- click SHELTER on the left nav
- click SEARCH ANIMALS in the shelter page links list
- see all the animals listed
- animals with uploaded photos should show their photos, be it front or side
- animals with no photos will have a new icon based on their species.
- the same can be verified on the owner details view, building room details view, and animal room assignment form.

## other notes
Did a little chipping away at test coverage in here as well.

closes #113
